modin/logging/logger_decorator.py

```diff
@@ -140,13 +145,18 @@ def run_and_log(*args: Tuple, **kwargs: Dict) -> Any:
             -------
             Any
             """
+            start_time = perf_counter()
             if LogMode.get() == "disable":
-                return obj(*args, **kwargs)
+                result = obj(*args, **kwargs)
+                emit_metric(metric_name, perf_counter() - start_time)
+                return result
 
             logger = get_logger()
             logger.log(log_level, start_line)
             try:
                 result = obj(*args, **kwargs)
+                if MetricsMode.get() == "enable":
+                    emit_metric(metric_name, perf_counter() - start_time)
```

Comment:
```suggestion
                emit_metric(metric_name, perf_counter() - start_time)
```

We already check `MetricsMode` inside `emit_metric`.